fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java

```diff
@@ -663,4 +673,82 @@ public LogicalPlan visitTableFunction(TableFunctionRelation node, ExpressionMapp
                 new ExpressionMapping(new Scope(RelationId.of(node), node.getRelationFields()), outputColumns)),
                 null, null);
     }
+
+    private Pair<OptExprBuilder, OptExprBuilder> processOnClauseSubquery(JoinRelation node, OptExprBuilder leftOpt,
+                                                                         OptExprBuilder rightOpt,
+                                                                         ExpressionMapping expressionMapping) {
+        Subquery subquery = node.getOnPredicate().getSubquery();
```

Comment:
The method only support one CTE? May this be a problem?

Reply:
Yes, it is. Currently, predicate can only has at most one subquery, and I think this limitation canbe removed. Should I handle this in this pr or another pr?

Reply:
fine, you can test two subquery in on-cluase later